src/pages/iou/request/step/IOURequestStepParticipants.js

```diff
@@ -66,15 +102,18 @@ function IOURequestStepParticipants({
             // When a participant is selected, the reportID needs to be saved because that's the reportID that will be used in the confirmation step.
             selectedReportID.current = lodashGet(val, '[0].reportID', reportID);
         },
-        [reportID, transactionID],
+        [reportID, transactionID, iouType, participants, updateRouteParams],
     );
 
-    const goToNextStep = useCallback(() => {
-        const nextStepIOUType = numberOfParticipants.current === 1 ? iouType : CONST.IOU.TYPE.SPLIT;
-        IOU.setMoneyRequestTag(transactionID, '');
-        IOU.setMoneyRequestCategory(transactionID, '');
-        Navigation.navigate(ROUTES.MONEY_REQUEST_STEP_CONFIRMATION.getRoute(nextStepIOUType, transactionID, selectedReportID.current || reportID));
-    }, [iouType, transactionID, reportID]);
+    const goToNextStep = useCallback(
+        (isSplit) => {
```

Comment:
Is this param really needed? It seems like `iouType` gets updated from the route once we tab 'split' on a participant. Simply using `iouType` here seems to work fine. Can you double check pelase?

Reply:
I think it would be nice if we removed it, or find a way to make this `onFinish(true);` less ambiguous, it's not immediately obvious what the boolean is doing there.

Reply:
Basically this seems to work fine, but I might be missing something

<img width="1023" alt="Screenshot 2024-02-28 at 04 49 45" src="https://github.com/Expensify/App/assets/9680864/32bdef70-d70a-4b88-9f05-2e930c2c1e5f">

Reply:
It's to cover a case where the `iouType` is not updated yet when `goToNextStep` is called.

1. Add a participant to split. At this point, `iouType` is split
2. Press the participant. When `goToNextStep` is called, `iouType` is still split, but it should be a request.


> find a way to make this onFinish(true); less ambiguous,

`addParticipant` also has `isSplit` on the param, maybe we can pass the iou type instead of boolean?

```
const goToNextStep = useCallback(
    (selectedIouType) => {
        const isSplit = selectedIouType === CONST.IOU.TYPE.SPLIT;    

```

Reply:
> 1. Add a participant to split. At this point, iouType is split
> 2. Press the participant. When goToNextStep is called, iouType is still split, but it should be a request.

Nice, yeah I knew I was missing something :D.

>addParticipant also has isSplit on the param, maybe we can pass the iou type instead of boolean?

Yeah this looks better, thanks!

Reply:
Done